### PR TITLE
Add sgid notice when running on AFS

### PR DESCRIPTION
**Pull request by michaelkuhn**
@georgi-mania and I spent some time today debugging bootstrapping problems that were caused by her home directory being located on AFS (see #14425). I'm not sure whether this is the best approach but we should definitely at least print some helpful message in cases like this.

### CODE REVIEW

**Comment by michaelkuhn** [comment]
@alalazo ping

File: lib/spack/spack/package_prefs.py [hunk 1]
```diff
@@ -15,6 +17,8 @@
 _lesser_spec_types = {'compiler': spack.spec.CompilerSpec,
                       'version': VersionList}
 
+_show_sgid_warning = True
```
Comment (alalazo) [inline lib/spack/spack/package_prefs.py]: Can we avoid using a global by setting options to `warnings.warn`?

Comment (michaelkuhn) [reply lib/spack/spack/package_prefs.py]: That seems to work (thanks for the hint, I didn't know about `warnings.warn`). We now get one warning per package (better than before) but also a second warning per package from the forked build environment (not so nice but acceptable, I guess).

**Comment by michaelkuhn** [comment]
@alalazo ping

**Review by alalazo — APPROVED** [review]

**Pull request merged**